packages/functions/src/dedup.ts

```diff
@@ -270,10 +270,21 @@ function dedupMaterials(document: Document): void {
 	const materials = root.listMaterials();
 	const duplicates = new Map<Material, Material>();
 	const skip = new Set(['name']);
+	let keptMaterials = 0;
 
 	// Compare each material to every other material — O(n²) — and mark duplicates for replacement.
 	for (let i = 0; i < materials.length; i++) {
 		const a = materials[i];
+
+		// Check if any parent edge has the modifyChild attribute set to true in which case we do not want the material to be deduped
+		const keep = document.getGraph()
+			.listParentEdges(a)
+			.some((ref) => ref.getAttributes().modifyChild === true);
+		if(keep) {
+			keptMaterials += 1;
+			continue;
+		}
```

Comment:
We'll hit a similar problem if a material extension is animated. If your KHR_animation_pointer implementation handles those references, then we can detect them here as well:

```js
const graph = document.getGraph();
const keep = [material, ...material.listExtensions()]
  .some((prop) => graph.listParentEdges(prop)
    .some((ref) => ref.getAttributes().modifyChild === true));
if (keep) continue;
```

Reply:
... actually this needs to go one layer deeper for animation of KHR_texture_transform, since that extends TextureInfo. Feel free to leave that as a "TODO". 😅

Reply:
I added a TODO for now yeah :)

Reply:
Thanks!